app/uk/gov/hmrc/exports/services/mapping/goodsshipment/WarehouseBuilder.scala

```diff
@@ -17,33 +17,29 @@
 package uk.gov.hmrc.exports.services.mapping.goodsshipment
 
 import javax.inject.Inject
-import uk.gov.hmrc.exports.models.declaration.WarehouseIdentification
 import uk.gov.hmrc.exports.services.mapping.ModifyingBuilder
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment.Warehouse
 import wco.datamodel.wco.declaration_ds.dms._2.{WarehouseIdentificationIDType, WarehouseTypeCodeType}
 
-class WarehouseBuilder @Inject()() extends ModifyingBuilder[WarehouseIdentification, GoodsShipment] {
-  override def buildThenAdd(model: WarehouseIdentification, goodsShipment: GoodsShipment): Unit =
-    if (isDefined(model)) {
-      goodsShipment.setWarehouse(createWarehouse(model))
+class WarehouseBuilder @Inject()() extends ModifyingBuilder[String, GoodsShipment] {
+  override def buildThenAdd(warehouseIdentification: String, goodsShipment: GoodsShipment): Unit =
+    if (warehouseIdentification.nonEmpty) {
+      goodsShipment.setWarehouse(createWarehouse(warehouseIdentification))
     }
 
-  private def isDefined(warehouse: WarehouseIdentification): Boolean =
-    warehouse.identificationNumber.isDefined
-
-  private def createWarehouse(data: WarehouseIdentification): Warehouse = {
+  private def createWarehouse(warehouseIdentification: String): Warehouse = {
     val warehouse = new Warehouse()
 
-    val warehouseIdentificationType = data.identificationNumber.map(_.substring(0, 1)).getOrElse("")
-    val warehouseIdentificationNumber = data.identificationNumber.map(_.substring(1)).getOrElse("")
+    val LocationsType = warehouseIdentification.substring(0, 1)
```

Comment:
The same like previous. I think better headOption and tail :)

Reply:
Ah ah!
Done! Please review